### PR TITLE
fix: check balance before enabling approve/send button

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: david <david@umaproject.org>

This adds a `useBalance` function which just does some searching on `useBalances` under the hood. This also converts any possible constant addresses to the proper eth checksummed address so that looks ups will work.

### CODE REVIEW

File: src/state/connection.ts [hunk 1]
```diff
@@ -30,7 +26,7 @@ const connectionSlice = createSlice({
   reducers: {
     update: (state, action: PayloadAction<Update>) => {
       const { account, chainId, provider, signer } = action.payload;
-      state.account = getAddress(account) ?? state.account;
+      state.account = account ? getAddress(account) : state.account;
       state.provider = provider ?? state.provider;
```
Comment (Tulun) [inline src/state/connection.ts:30]: What's going on here? Is this just a passthrough?

**Review by Tulun — APPROVED** [review]
Just a quick question, otherwise LGTM

File: src/state/connection.ts (hunk 1 above)
Comment (daywiss) [reply src/state/connection.ts:30]: yea just passes through becuase account can be undefined

**Comment by daywiss** [comment]
moved this logic into useSend

**Pull request merged**